### DISCUSSION TITLE
Noob question: It is buggy when I use stencil, drag the shape to paper, and type words with TextEditor

### DISCUSSION THREAD

**Discussion by whateverLive** [clientIO/joint#2038] [Q&A]
Hi all,

I want to use TextEditor so I can type text in shapes, so I seek doc and copy and paste their code
https://resources.jointjs.com/docs/rappid/v3.6/ui.TextEditor.html

However, althought I can type in words, the words are not in correct sequence. For example, if I type "12345", then it may display "23451", or other random wrong sequence. 

I am pretty sure it is related to the code for autosizing, because if I don't use the code, there would not be above problem:

`function autoSize(element) {

    var view = paper.findViewByModel(element);
    var textVel = view.vel.findOne('text');
    // Use bounding box without transformations so that our auto-sizing works
    // even on e.g. rotated element.
    var bbox = textVel.getBBox();
    // 16 = 2*8 which is the translation defined via ref-x ref-y for our rb element.
    element.resize(bbox.width + 16, bbox.height + 16);
}

graph.on('change:attrs', function(cell) { autoSize(cell) });`

This bug can be easily reproduced in codepen in 2 minutes: https://codepen.io/jointjs/pen/KKBYXrg
Just copy and paste the official TextEditor code.

What I have tried:
1. I think it may be related to size transformation when shapes are dragged, but I am not sure. I also tried to make the size of shape unchange when dragging but there is no use. (Not sure I do the operation correctly)

**Comment by kumilingus** [comment]
> This bug can be easily reproduced in codepen in 2 minutes: https://codepen.io/jointjs/pen/KKBYXrg
Just copy and paste the official TextEditor code.

Could you please clone the CodePen and add the code?

**Reply by CH-coding-fire** [reply]
Hey I am using another account. 

Here is the codepen: https://codepen.io/CH-coding-fire/pen/ExpBgrx

Strangely that when I use my home's computer, there is no bug! But if I use my office's computer, it has bugs. 

Do you know any possible causes?

**Reply by kumilingus** [reply]
In which browser and OS do you see the errors?

**Reply by CH-coding-fire** [reply]
I was using Chrome in my office. I don't remember the version but it should not be old

I am also using Chrome now in my home's computer.

**Reply by CH-coding-fire** [reply]
Also, in the first example in :https://resources.jointjs.com/docs/rappid/v3.5/ui.TextEditor.html
There is a toolbar for users to choose color, font etc. 

Is that a component that we can use? Because I don't find the document mention it

**Reply by kumilingus** [reply]
And the operating system?

**Reply by CH-coding-fire** [reply]
both window 10

**Reply by whateverLive** [reply]
Update: I use the same piece of code in codepen (the link above) in office computer, the problem happened again. What is more strange is at least 3 of the computers in my office has the same problem.

I updated office's Chrome to the latest version, but the problem persist

**Reply by kumilingus** [reply]
You can try opening CodePen in an incognito window. Maybe there is some extension installed that affects the TextEditor.

**Reply by whateverLive** [reply]
Hi, I have tried that, no use :(

**Reply by kumilingus** [reply]
Hi @whateverLive there was an issue with the `TextEditor` we managed to fix. The patch will be available soon.

To mitigate the problem for now, try resizing the element synchronously:
```js
element.resize(bbox.width + 16, bbox.height + 16, { async: false });
```

**Reply by whateverLive** [reply]
> Hi @whateverLive there was an issue with the `TextEditor` we managed to fix. The patch will be available soon.
> 
> To mitigate the problem for now, try resizing the element synchronously:
> 
> ```js
> element.resize(bbox.width + 16, bbox.height + 16, { async: false });
> ```

Hi, I am extremely grateful. I tried that synchronous method in CodePen: https://codepen.io/whateverLive/pen/vYzEWMd
And it wonderfully work in my office's computer. I will later implement in my project's code. But one question, I am currently using jointJs+ 3.2. (or it is called Rappid? It is always confusing for me). Do I need to upgrade the whole JointJS to the latest version? I would worry that it may break the code.

**Reply by kumilingus** [reply]
Regarding the naming I suggest reading #1664.

There were lot of changes/improvements made to the `TextEditor` plugin since `v3.2` (see [changelog](https://changelog.jointjs.com/rappid/3.6.0/changelog)). It's better to upgrade to the latest. 

If this is not possible, you can reach out to us via the support email and we could send you instructions how to patch the problem in your version.
